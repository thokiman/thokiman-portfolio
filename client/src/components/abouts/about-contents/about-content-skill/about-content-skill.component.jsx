import useWindowDimensions from 'components/hooks/window-dimensions/useWindowDimensions.component';
import React from 'react';
import { connect } from 'react-redux';
import { Route, Switch, withRouter } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import {
  selectSkillEngineer,
  selectSkillTechnology,
  selectSkillVisualDesignCommunication,
} from '../../../../redux/about/about.selectors';
import { selectIsSideBarActive } from '../../../../redux/header/header.selectors';
import EngineerSkill from '../../about-skills/about-skill-engineer/about-skill-engineer.component';
import SkillHeader from '../../about-skills/about-skill-header/about-skill-header.component';
import TechnologySkill from '../../about-skills/about-skill-technology/about-skill-technology.component';
import VisualArtSkill from '../../about-skills/about-skill-visual-art/about-skill-visual-art.component';
import { SkillContainer } from './about-content-skill.styles';
import { measureSkillContainerProps } from './about-content-skill.utils.style';

export const AboutSkill = ({
  match: { url },
  visualDesignCommunication,
  engineer,
  technology,
  isSideBarActive,
}) => {
  const { viewWidth, viewHeight } = useWindowDimensions();
  return (
    <SkillContainer
      className='skill-container'
      $measureskillcontainer={measureSkillContainerProps(viewWidth, viewHeight)}
      $issidebaractive={isSideBarActive}
    >
      <SkillHeader />
      <Switch>
        <Route
          exact
          path={`${url}`}
          render={() => <VisualArtSkill skill={visualDesignCommunication} />}
        />
        <Route
          path={visualDesignCommunication.routeName}
          render={() => <VisualArtSkill skill={visualDesignCommunication} />}
        />
        <Route
          path={engineer.routeName}
          render={() => <EngineerSkill skill={engineer} />}
        />
        <Route
          path={technology.routeName}
          render={() => <TechnologySkill skill={technology} />}
        />
      </Switch>
    </SkillContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  visualDesignCommunication: selectSkillVisualDesignCommunication,
  engineer: selectSkillEngineer,
  technology: selectSkillTechnology,
  isSideBarActive: selectIsSideBarActive,
});

export default withRouter(connect(mapStateToProps)(AboutSkill));
