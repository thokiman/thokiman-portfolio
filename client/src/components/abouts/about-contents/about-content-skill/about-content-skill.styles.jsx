import styled, { css } from 'styled-components';

const backgroundSkillContainerColor = 'rgba(18, 129, 148, 0.8)';
const maxWidthSmartphone = 800;
const AboutContentSkillIsNotActivated = css`
  display: none;
`;
const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return AboutContentSkillIsNotActivated;
  return null;
};

export const SkillContainer = styled.div`
  background-color: ${backgroundSkillContainerColor};
  width: 64.5vw;
  margin-right: 2vw;
  padding: 40px 0;
  height: 98%;
  font-family: baskerville-old-face;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100%;
    margin: 0;
    ${getSideBarActive}
  }
`;

SkillContainer.displayName = 'SkillContainer';
