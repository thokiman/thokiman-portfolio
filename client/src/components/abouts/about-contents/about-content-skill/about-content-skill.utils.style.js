import {
  hasScreenSize1,
  hasScreenSize1_1,
  hasScreenSize1_2,
  hasScreenSize1_3,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
  hasScreenSizeWindow,
  hasScreenSizeWindow_1,
} from '../../../../utils/common.component.utils';

export const measureSkillContainerProps = (viewWidth, viewHeight) => {
  const skillContainerOffset = 5;
  let skillContainerProps;

  if (hasScreenSizeWindow(viewWidth, viewHeight)) {
    console.log('about-content-skill', 1);
    skillContainerProps = {
      top: 44,
    };
  } else if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
    console.log('about-content-skill', 2);
    skillContainerProps = {
      top: 44,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('about-content-skill', 3);
    skillContainerProps = {
      top: 44,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('about-content-skill', 4);
    skillContainerProps = {
      top: 44,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('about-content-skill', 5);
    skillContainerProps = {
      top: 44,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('about-content-skill', 6);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('about-content-skill', 7);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('about-content-skill', 8);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('about-content-skill', 9);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('about-content-skill', 10);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('about-content-skill', 11);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('about-content-skill', 12);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('about-content-skill', 13);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('about-content-skill', 14);
    skillContainerProps = {
      top: 44 + skillContainerOffset,
    };
  } else {
    console.log('about-content-skill', 15);
    skillContainerProps = {
      top: 70,
    };
  }

  return skillContainerProps;
};
