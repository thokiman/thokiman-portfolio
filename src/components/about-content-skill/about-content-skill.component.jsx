import React from "react";
import { Switch, Route, withRouter } from "react-router-dom";
import { CSSTransitionGroup } from "react-transition-group";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  selectSkillEngineer,
  selectSkillTechnology,
  selectSkillVisualDesignCommunication,
} from "../../redux/about/about.selectors";
import SkillHeader from "../about-skill-header/about-skill-header.component";
import VisualArtSkill from "../about-skill-visual-art/about-skill-visual-art.component";
import EngineerSkill from "../about-skill-engineer/about-skill-engineer.component";
import TechnologySkill from "../about-skill-technology/about-skill-technology.component";
import { SkillContainer } from "./about-content-skill.styles";

const AboutSkill = ({
  match: { url },
  visualDesignCommunication,
  engineer,
  technology,
}) => {
  return (
    <SkillContainer className="skill-container">
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={600}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <SkillHeader />
        <Switch>
          <Route
            exact
            path={`${url}`}
            render={() => <VisualArtSkill skill={visualDesignCommunication} />}
          />
          <Route
            path={`${url}/visualart`}
            render={() => <VisualArtSkill skill={visualDesignCommunication} />}
          />
          <Route
            path={`${url}/engineer`}
            render={() => <EngineerSkill skill={engineer} />}
          />
          <Route
            path={`${url}/technology`}
            render={() => <TechnologySkill skill={technology} />}
          />
        </Switch>
      </CSSTransitionGroup>
    </SkillContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  visualDesignCommunication: selectSkillVisualDesignCommunication,
  engineer: selectSkillEngineer,
  technology: selectSkillTechnology,
});

export default withRouter(connect(mapStateToProps)(AboutSkill));
