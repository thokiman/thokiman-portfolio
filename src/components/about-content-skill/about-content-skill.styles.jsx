import styled from "styled-components";

export const SkillContainer = styled.div`
  background-color: rgba(18, 129, 148, 0.8);
  width: 64.5vw;
  margin-right: 2vw;
  padding: 40px 0;
  height: 98%;
  font-family: baskerville-old-face;
`;

SkillContainer.displayName = "SkillContainer";
