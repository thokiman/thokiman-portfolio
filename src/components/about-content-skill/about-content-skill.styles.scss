.skill-container {
    background-color: rgba($color: #128194, $alpha: 0.8);
    width: 69.9%;
    height: 98%;
    position: relative;
    right: 40px;
    font-family: baskerville-old-face;
}