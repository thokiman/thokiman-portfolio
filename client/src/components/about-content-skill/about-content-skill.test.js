import React from "react";
import { shallow } from "enzyme";

import { AboutSkill } from "./about-content-skill.component";

describe("AboutSkill component", () => {
  let wrapper;

  const mockObject = {};
  const mockIsSideBarHidden = false;
  const mockText = "";

  beforeEach(() => {
    const mockProps = {
      match: { url: mockText },
      visualDesignCommunication: mockObject,
      engineer: mockObject,
      technology: mockObject,
      isSideBarHidden: mockIsSideBarHidden,
    };
    wrapper = shallow(<AboutSkill {...mockProps} />);
  });

  it("should render AboutSkill component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
