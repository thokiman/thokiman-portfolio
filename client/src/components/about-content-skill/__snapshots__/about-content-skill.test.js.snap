// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutSkill component should render AboutSkill component 1`] = `ShallowWrapper {}`;
